var/spack/repos/builtin/packages/vtkm/package.py

```diff
@@ -17,46 +17,105 @@ class Vtkm(Package):
+            else:
+                options.append("-DVTKm_ENABLE_LOGGING:BOOL=OFF")
+
+            # mpi support
+            if "+mpi" in spec:
+                if self.spec.satisfies('@:1.2.0') and \
+                        spec['vtkm'].version.string != 'master':
+                    raise InstallError('mpi is not supported for\
+                            vtkm version lower than 1.3')
+                options.append("-DVTKm_ENABLE_MPI:BOOL=ON")
+            else:
+                options.append("-DVTKm_ENABLE_MPI:BOOL=OFF")
+
+            # openmp support
+            if "+openmp" in spec:
+                # openmp is added since version 1.3.0
+                if self.spec.satisfies('@:1.2.0') and \
+                        spec['vtkm'].version.string != 'master':
+                    raise InstallError('OpenMP is not supported for\
+                            vtkm version lower than 1.3')
+                # vtkm requires openmp no less than 4.0.0
+                if not spec.compiler.satisfies('gcc'):
+                    raise InstallError('Enabling OpenMP in vtkm requires gcc\
+                            compiler')
+                conflicts('%gcc@:4.8.9', None, 'The required OpenMP 4.0.0 is only supported with gcc no less than 4.9')
+                options.append("-DVTKm_ENABLE_OPENMP:BOOL=ON")
+            else:
+                options.append("-DVTKm_ENABLE_OPENMP:BOOL=OFF")
+
+            # rendering support
+            if "+rendering" in spec:
+                options.append("-DVTKm_ENABLE_RENDERING:BOOL=ON")
+            else:
+                options.append("-DVTKm_ENABLE_RENDERING:BOOL=OFF")
+
             # tbb support
             if "+tbb" in spec:
                 # vtk-m detectes tbb via TBB_ROOT env var
                 os.environ["TBB_ROOT"] = spec["tbb"].prefix
-                cmake_args.append("-DVTKm_ENABLE_TBB=ON")
-
-            # cuda support
-            if "+cuda" in spec:
-                cmake_args.append("-DVTKm_ENABLE_CUDA=ON")
-                # this fix is necessary if compiling platform has cuda, but
-                # no devices (this common for front end nodes on hpc clusters)
-                # we choose kepler as a lowest common denominator
-                cmake_args.append("-DVTKm_CUDA_Architecture=kepler")
+                options.append("-DVTKm_ENABLE_TBB:BOOL=ON")
+            else:
+                options.append("-DVTKm_ENABLE_TBB:BOOL=OFF")
 
             # use release, instead of release with debug symbols b/c vtkm libs
             # can overwhelm compilers with too many symbols
             for arg in std_cmake_args:
                 if arg.count("CMAKE_BUILD_TYPE") == 0:
-                    cmake_args.extend(std_cmake_args)
-            cmake_args.append("-DCMAKE_BUILD_TYPE=Release")
-            cmake(*cmake_args)
-            make()
-            make("install")
+                    options.extend(std_cmake_args)
+            options.append("-DCMAKE_BUILD_TYPE=Release")
```

Comment:
@HaochengLIU 

Note now that this extends `CMakePackage`: all subclasses of `CMakePackage` get a `build_type` variant by default. This will also set `-DCMAKE_BUILD_TYPE`. The default is `RelWithDebInfo`. You can override the variant to choose `Release` by default in this package:

```
variant('build_type', default='Release',
    description='CMake build type',
    values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
```